packages/stencil/components/input/src/components/osds-input/osds-input.e2e.ts

```diff
@@ -53,6 +53,108 @@ describe('e2e:osds-input', () => {
+  
+      // Click cross icon/button
+      const crossIcon = await page.find('osds-input >>> osds-icon[name="close"]');
+      expect(crossIcon).not.toBeNull();
+      await crossIcon.click();
+      await page.waitForChanges();
+  
+      // Verify input value is cleared
+      const value = await inputElement.getProperty('value');
+      expect(value).toBe('');
+    });
+  });
+
+  describe('attribute:hideable', () => {
+
+    it('should display hideable icon/button (eye closed)', async () => {
+      await setup({ attributes: { type: OdsInputType.text, value: 'Just ODS being ahead', hideable: true } });
+      
+      // Verify eye icon/button is visible
+      const eyeIcon = await page.find('osds-input >>> osds-icon[name="eye-closed"]');
+      expect(eyeIcon).not.toBeNull();
+    });
+
+    it('should display hideable icon/button (eye open)', async () => {
+      await setup({ attributes: { type: OdsInputType.password, value: 'Just ODS being ahead', hideable: true } });
+      
+      // Verify eye icon/button is visible
+      const eyeIcon = await page.find('osds-input >>> osds-icon[name="eye-open"]');
+      expect(eyeIcon).not.toBeNull();
+    });
+
+    it('should hide the input value when clicked', async () => {
+      // Setup component with clearable attribute and some initial value
+      await setup({ attributes: { type: OdsInputType.text, value: 'Just ODS being ahead', hideable: true } });
+  
+      // Click cross icon/button
+      const eyeIcon = await page.find('osds-input >>> osds-icon[name="eye-closed"]');
+      expect(eyeIcon).not.toBeNull();
+      await eyeIcon.click();
+      await page.waitForChanges();
+  
+      // Verify input value is cleared
+      const type = await inputElement.getProperty('type');
+      expect(type).toBe(OdsInputType.password);
+    });
+
+    it('should display the input value when clicked', async () => {
+      // Setup component with clearable attribute and some initial value
+      await setup({ attributes: { type: OdsInputType.text, value: 'Just ODS being ahead', hideable: true, masked: true } });
+  
+      // Click cross icon/button
+      const eyeIcon = await page.find('osds-input >>> osds-icon[name="eye-open"]');
+      expect(eyeIcon).not.toBeNull();
+      await eyeIcon.click();
+      await page.waitForChanges();
+  
+      // Verify input value is cleared
+      const type = await inputElement.getProperty('type');
+      expect(type).toBe(OdsInputType.text);
+    });
+  });
+
+  describe('attribute:masked', () => {
+
+    it('should hide input field content', async () => {
+      await setup({ attributes: { type: OdsInputType.text, value: 'Just ODS being ahead', masked: true } });
+      
+      // Verify eye icon/button is visible
+      const type = await inputElement.getProperty('type');
+      expect(type).toBe(OdsInputType.password);
+    });
+
+    it('should display input field content', async () => {
```

Comment:
description does not match the actual test

Reply:
should I rename it to "masked attribute should display input field content?"

Reply:
you're actually testing the type of the input here not the actual DOM impact, so more
```
it('should change input type to text when masked is set')
```